experimental/dds/tree2/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -217,9 +217,36 @@ export class ModularChangeFamily
 						// composition all the changes already reflected on the tree, but that is not something we
 						// care to support at this time.
 						if (!innerDstMap.has(id)) {
+							// Check for earlier destroys that this build might cancel-out with.
+							const didCancelOut = deleteFromNestedMap(
+								allDestroys,
+								setRevisionKey,
+								id,
+							);
+							if (!didCancelOut) {
+								innerDstMap.set(id, tree);
+							}
+						}
+					}
+					if (innerDstMap.size === 0) {
+						allBuilds.delete(setRevisionKey);
+					}
+				}
+			}
+			if (change.destroys) {
+				for (const [revisionKey, innerMap] of change.destroys) {
+					const setRevisionKey = revisionKey ?? revision;
+					const innerDstMap = getOrAddInMap(allDestroys, setRevisionKey, new Map());
+					for (const [id, tree] of innerMap) {
+						// Check for earlier builds that this destroy might cancel-out with.
+						const didCancelOut = deleteFromNestedMap(allBuilds, setRevisionKey, id);
+						if (!didCancelOut) {
 							innerDstMap.set(id, tree);
```

Comment:
Isn't the type for the values of `innerDstMap` `undefined`? Why can we set a value to a build tree?

Reply:
`tree` was undefined in that code, but it turns out that the typing was `Map<any, any>` anyway, so it would have let us do anything. It seems `getOrAddInMap` leads to very loose typing when passing in `new Map()`. I don't know how to fix the typing being so loose but I changed the usages here to pass in a map with more specific types which helps.